### ISSUE TITLE
Can I use images with not object to train model

### ISSUE THREAD

**Issue by bzhong2** [experiencor/keras-yolo3#122]
When I trained a model, I found that sometimes the models find some areas which are actually not the object. Can I use a lot of images without any object to train the model not to misclassify the background as the target objects? With the code give an error if the label has not object inside?

**Comment by anuar12** [comment]
Yes I think so. If you look at the loss, it has `(1-object_mask)` part in `YoloLayer.call()` (which corresponds to the no_obj part of the original loss in YOLO v1). So yes, loss will be non-zero for FPs.

**Comment by juncongfei** [comment]
@bzhong2 Hi, I'm also interested in your issue. Have you figure out how to train with negative samples?

**Comment by TiongSun** [comment]
How should the negative samples be annotated? I tried 0,0,0,0 but gen_anchor.py gives funny anchors:


average IOU for 9 anchors: nan
0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 71,87, 160,198

Any ideas?

**Comment by bzhong2** [comment]
@experiencor  It looks like this issue is interested for a couple of people but I do not know how to do it. Do you think your code has this feature inside somewhere?

**Comment by juncongfei** [comment]
@TiongSun For negative samples in my dataset, there are no object annotation. I change line 60  in voc.py so negative samples will not be filtered.

**Comment by anuar12** [comment]
I haven't implemented anything, you might need to add little tweaks in the data pre-processing code but in principle the architecture and general concept works the same. If you look closely in `__getitem__()` in `generator.py`, the label (yolo_1, etc..., which is 5D tensor) will be just zeros which would mean there is no object there.
